### ISSUE TITLE
got different result when use groupby().parallel_apply() compared to pandas

### ISSUE THREAD

**Issue by universewill** [nalepae/pandarallel#45]
Here is the code, pandarallel went error when it start to get the final result:
```
import pandas as pd

df = pd.DataFrame([['r', 1],['r', -1], ['r', 3], ['e', 10], ['r', 5], ['e', 4], ['t', 6], ['y', 8],
                   ['re', 11],['re', -1], ['rv', 3], ['ev', 10], ['rg', 5], ['eb', 4], ['tt', 3], ['tt', 7]], columns=['a','b'])
print(df)


def process_data(b_df):
    print(b_df.shape[0])
    df_list = []

    max_b = -1
    i = 0
    for index, row in b_df.iterrows():
        max_b = max(max_b, row['b'])
        if i==0:
            df_list.append(row['b'])
        i+=1

    df_list.append(max_b)

    df = pd.DataFrame([df_list], columns=['bb', 'max_b'])
    print(df)
    print(df_list)
    return df


train_dataframe = df.groupby(df['a']).apply(process_data)
print(' pandas process result ....')
print(train_dataframe)

from pandarallel import pandarallel
pandarallel.initialize(shm_size_mb=int(1e3), nb_workers=4)
train_dataframe0 = df.groupby('a').parallel_apply(process_data)
print(' pandarallel process result ....')
print(train_dataframe0)
```

**Comment by universewill** [comment]
Also found that pandas an pandarallel got different results when the func in  groupby. parallel_apply(func) return multi values or list or dataframe.

Example:
```
    def process(df):
        print(df.shape[0])
        df_list = []

        max_b = -1
        i = 0
        for index, row in df.iterrows():
            max_b = max(max_b, row['b'])
            if i==0:
                df_list.append(row['c'])
            i+=1

        df_list.append(max_b)

        # df = pd.DataFrame([df_list], columns=['cc', 'max_b'])
        return df_list

    df_list = [['batch_apply_' + str(random.randint(0, 20)), i+5, [i+6, str(i+7)]] for i in range(1000)]
    data_df = pd.DataFrame(df_list, columns=['a','b', 'c'])
    print('data_df:', data_df.count())
    print(data_df.head(5))
    #
    from pandarallel import pandarallel
    pandarallel.initialize(shm_size_mb=int(1e3), nb_workers=4)

    # newdatadf= data_df.groupby('a').parallel_apply(lambda x:process(x))
    newdatadf = data_df.groupby('a').apply(lambda x: process(x))

    print('newdatadf:')
    # newdatadf.reset_index()
    print(newdatadf.count())
    print(newdatadf.head(50))
```

**Comment by universewill** [comment]
Any help here? Many thanks! @nalepae @YashSinha1996

**Comment by kubapok** [comment]
I also got different results running my code with apply and parallel_apply.

**Comment by edoson** [comment]
Mmm.. this is not good at all..

**Comment by nalepae** [comment]
Fixed in `pandarallel v1.4.5` (thanks @bjvanderweij).

Please reopen if this issue still occurs.
